### ISSUE TITLE
Openshift 4: bucket creation error

### ISSUE THREAD

**Issue by AndrienkoAleksandr** [minio/operator#1346]
## Expected Behavior
Should be pre-created bucket defined in the tenant CR.

## Current Behavior
Operator writes in the logs about successfully created bucket, but that's not truth. 

## Steps to Reproduce (for bugs)
I recorded video with reproduction: https://youtu.be/XYymiP4WymU
1. Install minio operator using Openshift ui or subscription:

```yaml
---
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: minio-operator
  namespace: openshift-operators
spec:
  channel: stable
  installPlanApproval: Automatic
  name: minio-operator
  source: certified-operators
  sourceNamespace: openshift-marketplace
```

2. Create tenant cr and use another parts from https://github.com/minio/operator/tree/v4.5.4/examples/kustomization/base:

```yaml
apiVersion: minio.min.io/v2
kind: Tenant
metadata:
  name: storage
  namespace: minio-tenant
  ## Optionally pass labels to be applied to the statefulset pods
  labels:
    app: minio
  annotations:
    prometheus.io/path: /minio/v2/metrics/cluster
    prometheus.io/port: "9000"
    prometheus.io/scrape: "true"
spec:
  exposeServices:
    console: true
    minio: true
  features:
    bucketDNS: false
    domains: { }
  buckets:
    - name: "test"
      region: "us-east-1"
      objectLock: true
  certConfig: { }
  podManagementPolicy: Parallel
  configuration:
    name: storage-configuration
  env: [ ]
  serviceMetadata:
    minioServiceLabels: { }
    minioServiceAnnotations: { }
    consoleServiceLabels: { }
    consoleServiceAnnotations: { }
  priorityClassName: ""
  externalCaCertSecret: [ ]
  externalCertSecret: [ ]
  externalClientCertSecrets: [ ]
  image: quay.io/minio/minio:RELEASE.2022-09-17T00-09-45Z
  imagePullSecret: { }
  mountPath: /export
  subPath: ""
  serviceAccountName: ""
  pools:
    - servers: 1
      name: pool-0
      volumesPerServer: 2
      nodeSelector: { }
      tolerations: [ ]
      affinity:
        nodeAffinity: { }
        podAffinity: { }
        podAntiAffinity: { }
      resources: { }
      volumeClaimTemplate:
        apiVersion: v1
        kind: persistentvolumeclaims
        metadata: { }
        spec:
          accessModes:
            - ReadWriteOnce
          resources:
            requests:
              storage: 1Gi
        status: { }
      securityContext: {}
  requestAutoCert: true
```

## Regression
I don't know.

## Your Environment

* Version used (`minio-operator`): v4.5.4
* Environment name and version: Openshift 4.11
* Server type and version:
* Operating System and version (`uname -a`): Darwin Oleksandrs-MacBook-Pro 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct 9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64 x86_64 i386 Darwin
* Link to your deployment file:

**Comment by allanrogerr** [comment]
Using the latest `minio-operator` `v5.0.9`, I was able to create a bucket using the tenant yaml snippet. The steps followed are in: https://github.com/minio/operator/blob/master/testing/deploy-tenant.sh

The only yaml modification was to add the snippet to https://github.com/minio/operator/blob/master/examples/kustomization/tenant-lite/tenant.yaml

Snippet
```
spec:
  buckets:
    - name: "test"
      region: "us-east-1"
      objectLock: true
```

```
kubectl -n tenant-lite get tenant/myminio -o json | jq .spec.buckets
```
Once the tenant was running, this was the output:
```
[
  {
    "name": "test",
    "objectLock": true,
    "region": "us-east-1"
  }
]
```

**Comment by allanrogerr** [comment]
Could you retry with `minio-operator` v5.0.9 and let us know if this is still not working for you?

**Comment by allanrogerr** [comment]
It seems this is only available in the `master` branch. I will discuss internally to get this released soon.
See https://github.com/minio/operator/pull/1783

**Comment by allanrogerr** [comment]
This is now available in v5.0.10+.
